pkg/apis/minio.min.io/v2/types.go

```diff
@@ -658,7 +658,7 @@ type Pool struct {
 	//
 	// If provided, each pod on the Statefulset will run with the specified RuntimeClassName, for more info https://kubernetes.io/docs/concepts/containers/runtime-class/
 	// +optional
-	RuntimeClassName *string `json:"runtimeClassName,omitempty"`
+	RuntimeClassName string `json:"runtimeClassName,omitempty"`
```

Comment:
wouldn't the pointer be better to mark absence of value?

Reply:
Not really, absence of value could be indicated simply as an empty string "", or not including the field in the yaml. Also, pointers a NPE prone which is another reason I would like to avoid its usage for a simple string.

If I keep the *string definition, then I run into clumsy code like this: https://github.com/minio/console/pull/2565/files/d4208ed54152207ecaf4ad4018b8264becc1c42b#r1067450509

Reply:
The problem with using a string and not a string is that if the value is left empty, the k8s will always return this field with "" in it, by making it nullable we can have this field not be retuned unless it has an explicit value

Reply:
Thanks for pointing this out! This was also tested - see screenshot below. " Runtime Class Name" is non-existent unless it is explicitly defined in tenant. Not even "" shows up. Let me know your thoughts!
<img width="797" alt="image" src="https://user-images.githubusercontent.com/16472240/212100733-994d0c4d-dcde-4cc8-8f3d-e0f658cce157.png">

Reply:
This is k8s v1.25.3 on kind

Reply:
I spoke to @dvaldivia on this. His opinion was that we have so many fields in the CRD that using pointers is better so the k8s API returns simpler yamls.
Closing the current PR.